From 373efd40d226d11c49fe5d1a77dbef257edc7e38 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 22 Mar 2020 17:30:03 +0200
Subject: [PATCH] [SW-0] udelay: add volatile to counter

We have observed a different behavior of the new compiler.
It removed the udelay function code, and only return directive left.
This apparently happened because the compiler thought the function has
no effect, which is true regarding the data, but not the execution time.
Add volatile to the counter variable to prevent such a compiler
behavior.
Later we might want to change the function to use timers and thus also
raise its precision.

Change-Id: I5445c16bab0a64ff23742d6ea2539f27ca1a7dea
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index a82c2c6..6afc75a 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -27,7 +27,7 @@
 
 void hl_pll_udelay(unsigned int usec)
 {
-	u64 cnt = (cycles_per_usec * usec) / MIN_CYCLES_PER_ITER;
+	volatile u64 cnt = (cycles_per_usec * usec) / MIN_CYCLES_PER_ITER;
 
 	while (cnt--)
 		;
